port/unix/omrsysinfo.c

```diff
@@ -3191,6 +3212,40 @@ freeCgroupEntries(struct OMRPortLibrary *portLibrary, OMRCgroupEntry *cgEntryLis
 }
 
 /**
+ * Returns cgroup subsystems available for port library to use based on the
+ * subsystem entries in cgEntryList
+ *
+ * @param[in] portLibrary pointer to OMRPortLibrary
+ * @param[in] cgEntryList pointer to OMRCgroupEntry representing a circular list
+ * containing entry for each subsystem
+ *
+ * @return bitwise-OR of flags of type OMR_CGROUP_SUBSYSTEM_* indicating the
+ * subsystems that are available
+ */
+static uint64_t
+getAvailableSubsystems(struct OMRPortLibrary *portLibrary, OMRCgroupEntry *cgEntryList)
+{
+	OMRCgroupEntry *cgEntry = cgEntryList;
+	uint64_t availableSubsystems = 0;
+
+	if (NULL == cgEntry) {
+		goto _end;
+	}
+
+	do {
+		if (!strcmp(cgEntry->subsystem, subsystemNames[CPU])) {
+			availableSubsystems |= OMR_CGROUP_SUBSYSTEM_CPU;
+		} else if (!strcmp(cgEntry->subsystem, subsystemNames[MEMORY])) {
+			availableSubsystems |= OMR_CGROUP_SUBSYSTEM_MEMORY;
+		}
+		cgEntry = cgEntry->next;
+	} while (cgEntry!= cgEntryList);
```

Comment:
How long is this list on a typical system?  Vanilla hw vs inside a container?  If the list is typically capped at 2-3 entries, this code is fine.  If we expect longer lists, we'll want to write this differently, especially if it gets called at startup

Reply:
Currently `cgEntryList` has all cgroup subsystems configured in the system. In cgroup v1, that comes to 12 subsystems.
We can probably restrict it to only the ones we need - memory and cpu for now.
Let me add another change set on top of existing one which would highlight these changes.

Reply:
@DanHeidinga please take a look at commit 175daad that avoids adding every subsystem to `PPG_cgroupEntryList` and also sets `PPG_cgroupSubsystemsAvailable` more efficiently than before.